### PR TITLE
Fixed C# project module "AGS.CScript.Compiler" so that it successfully builds in xBuild on *nix systems

### PR DESCRIPTION
**Pull request by persn**
Needs #396 to be resolved in order to test properly since AGS.Cscript.Compiler references AGS.Types

### CODE REVIEW

File: Editor/AGS.CScript.Compiler/CodeBlockCompiler.cs [hunk 1]
```diff
@@ -269,7 +268,7 @@ private void GenerateCodeForExpressionWithoutOperator(Expression expression)
                 {
                     throw new CompilerMessage(ErrorCode.ParentIsNotAStruct, "'" + firstToken.Name + "' is not a struct");
                 }
-                memberName = reader.ReadNextToken();
+                ////Token memberName = reader.ReadNextToken();
```
Comment (ivan-mogilko) [inline Editor/AGS.CScript.Compiler/CodeBlockCompiler.cs]: I cannot tell if this is correct. ReadNextToken may advance some iterator, thus commenting its call out may cause further  operations break.

I suggest leaving original variable declaration (above) instead, and doing something like this on line 271:

> reader.ReadNextToken(); // memberName

File: Editor/AGS.Editor/Components/HelpCommandsComponent.cs [hunk 2]
```diff
@@ -325,10 +325,6 @@ private VersionCheckStatus CompareSoftwareVersions(string serverVersionText)
         private bool AdjustHelpWindowSize()
         {
             IntPtr helpHandle = Process.GetCurrentProcess().MainWindowHandle; // get the Help window handle
-            if (helpHandle == null)
-            {
-                return false;
-            }
```
Comment (ivan-mogilko) [inline Editor/AGS.Editor/Components/HelpCommandsComponent.cs]: I wonder if Chris Jones meant to compare this with 0 (zero handle) instead?

**Comment by ivan-mogilko** [comment]
I checked this out, and found just two changes that need to be reworked in my opinion.

Regarding unused _isLegacyString flag, I **do** know what that means, but since this compiler is not complete, I cannot tell how it was supposed to be used. If anyone would finish this in the future, they would have to address this issue anyway (assuming legacy strings still would be supported, which is questionable).

**Comment by ivan-mogilko** [comment]
E: actually, I rejected one of my comments, noticing that it only affects local reader object.

**Comment by persn** [comment]
Updated according to feedback and rebased on master.

**Pull request merged**